### PR TITLE
docs(rfc): update packaging with shipping types rfc

### PR DESCRIPTION
**Pull request by Hotell**
#### Pre-requisite: 
- [x] https://github.com/microsoft/fluentui/pull/22468

Adds shipping types RFC chapter

[Preview 👀](https://github.com/microsoft/fluentui/pull/22469/files?short_path=da942a8#diff-da942a8b306a106a49f511ada3834c66929e8adb5ec1202ce7b24d229fa8d836)

## Related Issue(s)

Fixes partially https://github.com/microsoft/fluentui/issues/22429

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/65897f7924362a8c8bf581ddca162c8ed4247011'>65897f7924362a8c8bf581ddca162c8ed4247011</a></sub>
<div id="bundleSizeComment9423"/>

File: rfcs/shared/build-system/03-packaging-for-npm.md [hunk 1]
```diff
@@ -164,17 +172,89 @@ This approach significantly improves our current situation:
   - non explicit for human (if `files` property is used, some files will be automatically and always shipped)
   - 2 sources of "truth" (`.npmignore` and `files`)
 
-### 2. shipping only type definition rollup
+### 2. shipping type declarations (TypeScript)
 
 > **NOTE:** to make this RFC more focused this will be covered in followup PR's if needed
```
Comment (andrefcdias) [inline rfcs/shared/build-system/03-packaging-for-npm.md:177]:
```suggestion
```

File: rfcs/shared/build-system/03-packaging-for-npm.md [hunk 2]
```diff
@@ -164,17 +172,89 @@ This approach significantly improves our current situation:
...
+- opportunity to leverage TypeScript language constructs beyond consumers support
+- less files shipped over network
+- bit faster build process
 
 #### Cons
 
-- TBA
+- additional configuration to maintain within our legacy tool chain
 
 ### 3. shipping only rollup-ed(bundled) implementation files
```
Comment (andrefcdias) [inline rfcs/shared/build-system/03-packaging-for-npm.md:259]: This solution is what's mentioned as an approach for problem 1 of section 2, right?

image

**Review by andrefcdias — APPROVED** [review]
Both approaches seem good to me 👌

3 (shipping only rollup-ed(bundled) implementation files) and 4 (package.json post-processing) won't be tackled in this PR, correct?

File: rfcs/shared/build-system/03-packaging-for-npm.md (hunk 2 above)
Comment (Hotell) [reply rfcs/shared/build-system/03-packaging-for-npm.md:259]: no, there is nothing regarding implementation files in this PR/RFC

**Review by spmonahan — APPROVED** [review]
Looks great!

**Review by khmakoto — APPROVED** [review]
I think this makes sense, I especially like not exposing internals if we don't need to.

File: rfcs/shared/build-system/03-packaging-for-npm.md [hunk 3]
```diff
@@ -172,25 +172,88 @@ This approach significantly improves our current situation:
...
+export declare const bodyClassName = 'fui-Body';
+export declare const Body: FunctionComponent<TextWrapperProps>;
+```
 
 #### Detailed Design or Proposal
 
-- TBA
+**Problem 1 solution:**
+
+- ship only [rolluped declaration files `.d.ts`](https://medium.com/@martin_hotell/typescript-library-tips-rollup-your-types-995153cc81c7) for all public API entrances. We can leverage already existing setup via api-extractor.
```
Comment (ling1726) [inline rfcs/shared/build-system/03-packaging-for-npm.md:223]: The diagram mentions rollup, but the solution using api-extractor is still comptible with how we currently build with tsc right ?

**Review by ling1726 — APPROVED** [review]

**Pull request merged**